lib/Service/AvatarService.php

```diff
@@ -81,7 +85,40 @@ public function setAvatarFromRequest(Room $room, ?array $file): void {
 		$this->setAvatar($room, $image);
 	}
 
-	public function setAvatar(Room $room, \OC_Image $image): void {
+	public function setAvatarFromEmoji(Room $room, string $emoji, ?string $color): void {
+		if ($room->getType() === Room::TYPE_ONE_TO_ONE || $room->getType() === Room::TYPE_ONE_TO_ONE_FORMER) {
+			throw new InvalidArgumentException($this->l->t('One-to-one rooms always need to show the other users avatar'));
+		}
+
+		if ($this->getFirstCombinedEmoji($emoji) !== $emoji) {
+			throw new InvalidArgumentException($this->l->t('Invalid emoji character'));
+		}
+
+		if ($color === null) {
+			$color = self::THEMING_PLACEHOLDER;
+		} elseif (!preg_match('/^[a-fA-F0-9]{6}$/', $color)) {
+			throw new InvalidArgumentException($this->l->t('Invalid background color'));
+		}
+
+		$content = $this->getEmojiAvatar($emoji, $color);
+		$this->setSVGAvatar($room, $content);
+	}
+
+	public function setSVGAvatar(Room $room, string $svgContent): void {
+		$token = $room->getToken();
+		$avatarFolder = $this->getAvatarFolder($token);
+
+		// Delete previous avatars
+		foreach ($avatarFolder->getDirectoryListing() as $file) {
+			$file->delete();
+		}
+
+		$avatarName = $this->random->generate(16, ISecureRandom::CHAR_HUMAN_READABLE) . '.svg';
+		$avatarFolder->newFile($avatarName, $svgContent);
+		$this->roomService->setAvatar($room, $avatarName);
+	}
+
+	public function setAvatar(Room $room, \OC_Image $image, bool $allowSvg = false): void {
```

Comment:
Maybe other name then my suggestion would be best but the idea is to rename this method because isn't more the unique method to set avatar.
```suggestion
	public function setAvatarFromBinary(Room $room, \OC_Image $image, bool $allowSvg = false): void {
```

Reply:
The thing is we can't have a \OC_Image as it does not support SVG. And then the differences are so much, that I'm not sure it's worth to enforce them into 1 method

Reply:
My suggestion isn't to use only a method, is only to rename the actual to `setAvatarFromBinary` and maintain the new as `setSVGAvatar`. Only to don't have a generic method called `setAvatar`